### ISSUE TITLE
[MJLINK-42] NullPointerException when running jlink:jlink

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-jlink-plugin#439]
**[Dominique Gunia](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=dg76)** opened **[MJLINK-42](https://issues.apache.org/jira/browse/MJLINK-42?redirect=false)** and commented

Hi!

I am getting a NullPointerException when running jlink:jlink:

```java
Execution default-cli of goal org.apache.maven.plugins:maven-jlink-plugin:3.0.0-alpha-1:jlink failed.
 at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:224)
 at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:153)
 at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:145)
 at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:116)
 at org.apache.maven.lifecycle.internal.LifecycleModuleBuilder.buildProject(LifecycleModuleBuilder.java:80)
 at org.apache.maven.lifecycle.internal.builder.singlethreaded.SingleThreadedBuilder.build(SingleThreadedBuilder.java:51)
 at org.apache.maven.lifecycle.internal.LifecycleStarter.execute(LifecycleStarter.java:120)
 at org.apache.maven.DefaultMaven.doExecute(DefaultMaven.java:347)
 at org.apache.maven.DefaultMaven.execute(DefaultMaven.java:154)
 at org.apache.maven.cli.MavenCli.execute(MavenCli.java:584)
 at org.apache.maven.cli.MavenCli.doMain(MavenCli.java:213)
 at org.apache.maven.cli.MavenCli.main(MavenCli.java:157)
 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
 at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
 at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
 at java.base/java.lang.reflect.Method.invoke(Method.java:566)
 at org.codehaus.plexus.classworlds.launcher.Launcher.launchEnhanced(Launcher.java:289)
 at org.codehaus.plexus.classworlds.launcher.Launcher.launch(Launcher.java:229)
 at org.codehaus.plexus.classworlds.launcher.Launcher.mainWithExitCode(Launcher.java:415)
 at org.codehaus.plexus.classworlds.launcher.Launcher.main(Launcher.java:356)
Caused by: org.apache.maven.plugin.PluginExecutionException: Execution default-cli of goal org.apache.maven.plugins:maven-jlink-plugin:3.0.0-alpha-1:jlink failed.
 at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:143)
 at org.apache.maven.lifecycle.internal.MojoExecutor.execute(MojoExecutor.java:208)
 ... 19 more
Caused by: java.lang.NullPointerException
 at org.codehaus.plexus.languages.java.jpms.ResolvePathsRequest$1.toPath(ResolvePathsRequest.java:52)
 at org.codehaus.plexus.languages.java.jpms.ResolvePathsRequest$1.toPath(ResolvePathsRequest.java:48)
 at org.codehaus.plexus.languages.java.jpms.LocationManager.resolvePaths(LocationManager.java:109)
 at org.apache.maven.plugins.jlink.JLinkMojo.preparePaths(JLinkMojo.java:347)
 at org.apache.maven.plugins.jlink.JLinkMojo.execute(JLinkMojo.java:264)
 at org.apache.maven.plugin.DefaultBuildPluginManager.executeMojo(DefaultBuildPluginManager.java:132)
 ... 20 more
```

Is this a known problem? How can it be fixed?

Thank you very much! :)

 

---

**Affects:** 3.0.0-alpha-1

**Issue Links:**
- [MJLINK-4](https://issues.apache.org/jira/browse/MJLINK-4) NPE on execution
 (_**"duplicates"**_)

4 votes, 8 watchers

**Comment by jira-importer** [comment]
**[Thorsten Heit](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=t_heit)** commented

The same here under Windows 8.1 x64:

```
openjdk version "13.0.2" 2020-01-14
OpenJDK Runtime Environment AdoptOpenJDK (build 13.0.2+8)
OpenJDK 64-Bit Server VM AdoptOpenJDK (build 13.0.2+8, mixed mode, sharing)
```

My project structure is basically the following:

```
- main
  +- mod-src
  +- mod-app
  \- mod-jlink 
```

The mod-src contains the source code. mod-app depends on mod-src and uses m-assembly-p and appassembler-maven-plugin to create a .tar.gz-version of the source that can be started via command line.

mod-jlink depends on mod-src and is of packaging type jlink as described in the docs.

**Comment by jira-importer** [comment]
**[Leif Gruenwoldt](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=leif81)** commented

I was getting this too until I changed maven-compiler-plugin configuration to source & target from 8 to 11.

Edit: Never mind. I'm still getting the null pointer.

**Comment by jira-importer** [comment]
**[Michael Ellis](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael51773)** commented

I have the same issue.

**WHEN BUILDING FROM THE COMMAND LINE**

However, when issue the **mvn install** install command from the top level of my modular project, it works, but when I cs into the **mod-jlink** directory and perform the **mvn install** it fails with:

```

[INFO] Scanning for projects...

[INFO]

[INFO] -----------------< com.soebes.examples.jdk9:mod-jlink >-----------------

[INFO] Building mod-jlink 1.0-SNAPSHOT

[INFO] -------------------------------[ jlink ]--------------------------------

[INFO]

[INFO] — maven-jlink-plugin:3.0.0-alpha-1:jlink (default-jlink) @ mod-jlink —

[INFO] Toolchain in maven-jlink-plugin: jlink [ /Library/Java/JavaVirtualMachines/jdk-11.0.6.jdk/Contents/Home/bin/jlink ]

[INFO] ------------------------------------------------------------------------

[INFO] BUILD FAILURE

[INFO] ------------------------------------------------------------------------

[INFO] Total time:  0.845 s

[INFO] Finished at: 2020-06-05T11:28:12+01:00

[INFO] ------------------------------------------------------------------------

[ERROR] Failed to execute goal org.apache.maven.plugins:maven-jlink-plugin:3.0.0-alpha-1:jlink (default-jlink) on project mod-jlink: Execution default-jlink of goal org.apache.maven.plugins:maven-jlink-plugin:3.0.0-alp
```

**WHEN BUILDING FROM NETBEANS IDE**
I get the same failure whether building the overall main project or just the mod-jlink project.

Just discovered that I can fix things so the NetBeans can build the overall project by upgrading NetBeans to use 3.6.1 maven (rather than the maven that is bundled with Netbeans 11.3). However, just the command line invocation, NetBeans it fails to build if you explicitly select the mod-jlink project for building.

**Comment by jira-importer** [comment]
**[Andres Almiray](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=aalmiray)** commented

MCVE available at https://github.com/aalmiray/jlink-mvn-mcve

**Comment by jira-importer** [comment]
**[Benjamin Marwell](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=mampf86)** commented

>  https://github.com/aalmiray/jlink-mvn-mcve

Cannot reproduce, but I used Java 14-openj9: https://gist.github.com/bmarwell/6f39756780b05233da2c66b912bc2afd

Also works using JDK 13 (OpenJ9 from adoptopenjdk.net).

PS: You do not need to use `mvn install`. You can also use `mvn clean verify`.

**Comment by jira-importer** [comment]
**[Leif Gruenwoldt](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=leif81)** commented

I can reproduce with [https://github.com/aalmiray/jlink-mvn-mcve ](https://github.com/aalmiray/jlink-mvn-mcve) using Maven 3.6.3 and AdoptOpenJDK 11.0.5.

But the good news is when I modify the pom.xml in the MVCE to use maven-jlink-plugin 3.0.0 I don't see the error anymore. So, I think this can be closed and marked fixed in 3.0.0.
